qiita_pet/test/test_study_handlers.py

```diff
@@ -382,5 +382,36 @@ class TestEBISubmitHandler(TestHandlerBase):
     pass
 
 
+class TestDelete(TestHandlerBase):
+    def test_delete_sample_template(self):
+        response = self.post('/study/description/1',
+                             {'sample_template_id': 1,
+                              'action': 'delete_sample_template'})
+        self.assertEqual(response.code, 200)
```

Comment:
Concrete example [here](https://github.com/biocore/qiita/blob/master/qiita_pet/test/test_analysis_handlers.py#L86) Where the database change called for in the page is explicitly tested that it happens.